sources/api/migration/migrator/src/main.rs

```diff
@@ -116,27 +121,37 @@ where
 }
 
 pub(crate) async fn run(args: &Args) -> Result<()> {
-    let migrated_datastore = perform_migrations(args).await?;
+    let current_version = get_current_version(&args.datastore_path).await?;
+
+    exit_unless_version_change(&current_version, args).await?;
+
+    let migrated_datastore = perform_migrations(&current_version, args).await?;
 
     // Remove all the weak setting and all metadata
```

Comment:
Nit: do we also need to update the comments here now that we are clearing more resources.

Reply:
Nice catch. I think the readability here is strong enough that we can just delete the comment -- I'll do it.